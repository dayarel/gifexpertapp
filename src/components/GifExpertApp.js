import React, { useState } from "react";
import AddCategory from "./AddCategory";

import PropTypes from "prop-types";
import GifGrid from "./GifGrid";
import Popular from "./Popular";
const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <div className="container">
      <h2>Gif Expert</h2>
      <p>What do you like to find</p>
      <Popular setCategories={setCategories} />
      <AddCategory setCategories={setCategories} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </div>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default GifExpertApp;
