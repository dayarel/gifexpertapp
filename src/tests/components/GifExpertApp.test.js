import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../../components/GifExpertApp";

describe("Pruebas sobre GifExpertApp", () => {
  test("se renderiza el componente", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar una lista de categorias", () => {
    const categories = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    // expect(wrapper.find("GifGrid").length).toBe(categories.lenth);
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
