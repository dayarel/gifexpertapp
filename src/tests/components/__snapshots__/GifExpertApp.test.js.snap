// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifExpertApp debe de mostrar una lista de categorias 1`] = `
<div
  className="container"
>
  <h2>
    Gif Expert
  </h2>
  <p>
    What do you like to find
  </p>
  <Popular
    setCategories={[Function]}
  />
  <AddCategory
    setCategories={[Function]}
  />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
</div>
`;

exports[`Pruebas sobre GifExpertApp se renderiza el componente 1`] = `
<div
  className="container"
>
  <h2>
    Gif Expert
  </h2>
  <p>
    What do you like to find
  </p>
  <Popular
    setCategories={[Function]}
  />
  <AddCategory
    setCategories={[Function]}
  />
</div>
`;
